internal/transport/http2_client.go

```diff
@@ -589,7 +590,11 @@ func (t *http2Client) getTrAuthData(ctx context.Context, audience string) (map[s
 	for _, c := range t.perRPCCreds {
 		data, err := c.GetRequestMetadata(ctx, audience)
 		if err != nil {
-			if _, ok := status.FromError(err); ok {
+			if st, ok := status.FromError(err); ok {
```

Comment:
I'm a little concerned that we have to do this in multiple places. How can we guarantee that we haven't missed a place (in this PR), or that we don't miss one in the future when we are adding something.

Reply:
A54 specifies the places that should be covered, and this PR should cover all of them.  I've also reached out to Eric about whether other plugins (e.g. compressors and encoders) should be considered.